### PR TITLE
fix: call df_func with literal

### PR DESCRIPTION
**Pull request by discord9**
I hereby agree to the terms of the [GreptimeDB CLA](https://github.com/GreptimeTeam/.github/blob/main/CLA.md).

## What's changed and what's your intention?

fix a bug where calling datafusion function from flow with literal args will cause datafusion to throw error with unmatched type

Please explain IN DETAIL what the changes are in this PR and why they are needed:

- add a check to not rewrite literal in df func's calling arguments
- add sqlness test for that case

## Checklist

- [ ] I have written the necessary rustdoc comments.
- [ ] I have added the necessary unit tests and integration tests.
- [ ] This PR requires documentation updates.

## Summary by CodeRabbit

- **New Features**
  - Enhanced logging for scalar function evaluation, providing better insights during the debugging process.

- **Improvements**
  - Improved handling of literals in scalar functions with the introduction of a new method.

### CODE REVIEW

File: src/flow/src/expr/scalar.rs [hunk 1]
```diff
@@ -209,9 +210,11 @@ impl DfScalarFunction {
 
     // TODO(discord9): add RecordBatch support
     pub fn eval(&self, values: &[Value], exprs: &[ScalarExpr]) -> Result<Value, EvalError> {
+        info!("evaluating datafusion scalar function: {:?}", self);
+        info!("values: {:?}, exprs: {:?}", values, exprs);
```
Comment (WenyXu) [inline src/flow/src/expr/scalar.rs:214]: nit: use `debug!`?

File: src/flow/src/expr/scalar.rs [hunk 2]
```diff
@@ -209,9 +210,11 @@ impl DfScalarFunction {
...
         // first eval exprs to construct values to feed to datafusion
         let values: Vec<_> = Self::eval_args(values, exprs)?;
-
+        info!("evaluated values: {:?}", values);
```
Comment (WenyXu) [inline src/flow/src/expr/scalar.rs:217]: ditto

File: src/flow/src/expr/scalar.rs [hunk 3]
```diff
@@ -242,6 +245,10 @@ impl DfScalarFunction {
             }
             .build()
         })?;
+        info!(
+            "evaluating datafusion scalar function with record batch: {:?}",
+            rb
+        );
```
Comment (WenyXu) [inline src/flow/src/expr/scalar.rs:251]: The record batch could be very large.

File: src/flow/src/expr/scalar.rs [hunk 4]
```diff
@@ -296,6 +303,7 @@ impl RawDfScalarFn {
             .context(DecodeRelSnafu)
             .map_err(BoxedError::new)
             .context(crate::error::ExternalSnafu)?;
+        info!("Decoded scalar function: {:?}", f);
```
Comment (killme2008) [inline src/flow/src/expr/scalar.rs:306]: Abuse of info logs.

File: src/flow/src/transform/expr.rs [hunk 5]
```diff
@@ -144,9 +159,9 @@ impl TypedExpr {
     ) -> Result<TypedExpr, Error> {
         let (arg_exprs, arg_types): (Vec<_>, Vec<_>) =
             arg_exprs_typed.into_iter().map(|e| (e.expr, e.typ)).unzip();
-
+        info!("Before rewrite: {:?}", f);
```
Comment (killme2008) [inline src/flow/src/transform/expr.rs:162]: Pease use debug level.

**Review by zhongzc — APPROVED** [review]

**Review by WenyXu — APPROVED** [review]

**Pull request merged**